File: 217-contains-duplicate/contains-duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_map<int,bool> m;
        for(auto i:nums) {
            if(m.find(i)==m.end()) m[i]=0;
            else m[i]=1;
        }
        int c=0;
        for(auto i:m) if(i.second==0) c++;
        return c!=nums.size();
    }
};

File: 217-contains-duplicate/contains-duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        HashMap<Integer, Integer> m = new HashMap<Integer, Integer>();
        for(int i=0;i<nums.length;i++) {
            if(m.containsKey(nums[i])) return true;
            else m.put(nums[i],1);
        }
        return false;
    }
}